### PR TITLE
improve the documentation on initialPageSetting

### PR DESCRIPTION
**Pull request by ZheSun88**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/flow/3207)

### CODE REVIEW

**Comment by caalador** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3207#-L0xNK5Fytk5LbOJx6aN:-L0xNK5Fytk5LbOJx6aO:b-794vee) (no related file):*
Missing using inline in `PageConfigurator` and a sample for the inlining of a html file.
See. `BootstrapHandlerTest::InitialPageConfiguratorAppendFiles.class`

Also should say something about the possibility to use: `InitialPageSettings::addInlineWithContents`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ZheSun88** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3207#-L0xNK5Fytk5LbOJx6aN:-L0y9LVPBzsfyhbY-pw9:bz3hg40) (no related file):*
<details><summary><i>Previously, caalador wrote…</i></summary><blockquote>

Missing using inline in `PageConfigurator` and a sample for the inlining of a html file.
See. `BootstrapHandlerTest::InitialPageConfiguratorAppendFiles.class`

Also should say something about the possibility to use: `InitialPageSettings::addInlineWithContents`
</blockquote></details>

I have added the possible 4 options and code examples, and trying to highlight the using inline in PageConfigurator.
Should i do it in this way? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by caalador** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 126 at r2](https://reviewable.io:443/reviews/vaadin/flow/3207#-L0yHToC5-SQntP51H-j:-L0yHToC5-SQntP51H-k:b6sj9e2) ([raw file](https://github.com/vaadin/flow/blob/7666e3d08c7f143aad4a93a2c068a563a88782b5/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L126)):*
> ```AsciiDoc
> }
> ----
> There is another common case that you have html contents to be appended to your page, for an example you will have a `favicon`. There are a few ways of doing this:
> ```

Make it's own topic of this and add links to it from each of the other topics.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ahie** [comment]
Reviewed 1 of 2 files at r2.
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 210 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WIwzwpv9V7wATSFO5:-L1WIwzx45Dyb2Dly1z5:bnw9iku) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L210)):*
> ```AsciiDoc
> Now this listener should be added to a `ServiceInitEvent` which is sent when a Vaadin service is initialized. Take a look on the <<../miscellaneous/tutorial-service-init-listener#,ServiceInitListener tutorial>> on how to configure it.
> 
> == Adding static HTML contents
> ```

Shouldn't there be a newline after the title for this to render correctly?

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 259 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WJ4iLhPqVp92Na9v2:-L1WJ4iLhPqVp92Na9v3:b-8or7eg) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L259)):*
> ```AsciiDoc
> 
> But most commonly, `favicon` contains quite a lot small icons, in this case, you can see that you will easily do a lot of hard coding. To avoid this, we are *highly suggest* you to move all the html contents into a file, (e.g. `favicon.html`) and inline this file in your `PageConfigurator`
> [source,java]
> ```

Same here, newline above this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ZheSun88** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 210 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WIwzwpv9V7wATSFO5:-L1WL7lqk_oA6f9y6B5a:b-7rl0jr) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L210)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

Shouldn't there be a newline after the title for this to render correctly?
</blockquote></details>

done 

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 259 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WJ4iLhPqVp92Na9v2:-L1WL6hL4ZeIbudDvrve:b-7rl0jr) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L259)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

Same here, newline above this.
</blockquote></details>

done 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ahie** [comment]
Review status: 1 of 2 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 211 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WLDF4JFZUzmzGspux:-L1WLDF4JFZUzmzGspuy:b-73h4pl) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L211)):*
> There is another common case that you have html contents to be appended to your page, for an example you will have a `favicon`. There are a few ways of doing this:

I would reword this to:

> The framework provides multiple ways of adding static content to the page. Here we cover three different ways of adding a favicon.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 229 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WJXhKJFZaJSgOYfG3:-L1WJXhKJFZaJSgOYfG4:b-8or7eg) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L229)):*
> ```AsciiDoc
> 
>  - using `InitialPageSettings#addInlineWithContents()`
> [source,java]
> ```

Same here, newline above this.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 258 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WK1sAD82XsnA8qPHz:-L1WK1sAD82XsnA8qPI-:bdnzc1f) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L258)):*
> But most commonly, `favicon` contains quite a lot small icons, in this case, you can see that you will easily do a lot of hard coding. To avoid this, we are *highly suggest* you to move all the html contents into a file, (e.g. `favicon.html`) and inline this file in your `PageConfigurator`

A favicon is just an icon so the example of adding an html import doesn't really make sense. Just mention here that you can inline contents from a file and change the `favicon.html` to something else in the example.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ZheSun88** [comment]
Review status: 1 of 2 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 211 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WLDF4JFZUzmzGspux:-L1WPsFWi5yfVQTZsayL:bb7f2mf) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L211)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> There is another common case that you have html contents to be appended to your page, for an example you will have a `favicon`. There are a few ways of doing this:

I would reword this to:

> The framework provides multiple ways of adding static content to the page. Here we cover three different ways of adding a favicon.
</blockquote></details>

done

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 229 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WJXhKJFZaJSgOYfG3:-L1WPsyLT4dkGpSYTHiR:b-jctua3) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L229)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

Same here, newline above this.
</blockquote></details>

oops, sorry. done

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 258 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WK1sAD82XsnA8qPHz:-L1WPwNEeVMQDWnRyZHq:b-3hza9u) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L258)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> But most commonly, `favicon` contains quite a lot small icons, in this case, you can see that you will easily do a lot of hard coding. To avoid this, we are *highly suggest* you to move all the html contents into a file, (e.g. `favicon.html`) and inline this file in your `PageConfigurator`

A favicon is just an icon so the example of adding an html import doesn't really make sense. Just mention here that you can inline contents from a file and change the `favicon.html` to something else in the example.
</blockquote></details>

in fact, the main purpose for the document is emphasis this use case, we don't want the user to add a lot of code as it is shown https://github.com/vaadin/flow/issues/3163. 

I reworded the sentence to avoid the confusing `favicon` to `yourContent`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ahie** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 258 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WK1sAD82XsnA8qPHz:-L1WJu7w5QOqUPNtNoe5:b-9gyrwy) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L258)):*
<details><summary><i>Previously, ZheSun88 (Sun Zhe) wrote…</i></summary><blockquote>

in fact, the main purpose for the document is emphasis this use case, we don't want the user to add a lot of code as it is shown https://github.com/vaadin/flow/issues/3163. 

I reworded the sentence to avoid the confusing `favicon` to `yourContent`.
</blockquote></details>

Nit: `your-content.html` is more standard than `yourContent.html` 

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 260 at r4](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WRDsN-eR8fiS0cp_H:-L1WRDsN-eR8fiS0cp_I:b1dfc0k) ([raw file](https://github.com/vaadin/flow/blob/2d38ad3575525ce334a731073af292330547d63b/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L260)):*
> we are *highly suggest*

Would change to "we recommend".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ZheSun88** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 258 at r3](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WK1sAD82XsnA8qPHz:-L1WRuriQFj4yYSWyA-4:b9q8y3r) ([raw file](https://github.com/vaadin/flow/blob/984836052d5a4ae20052420edf935b87beeb8892/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L258)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

Nit: `your-content.html` is more standard than `yourContent.html` 
</blockquote></details>

oh, thanks.. done

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 260 at r4](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WRDsN-eR8fiS0cp_H:-L1WRwdpUpY_qx4ZU34Q:bb7f2mf) ([raw file](https://github.com/vaadin/flow/blob/2d38ad3575525ce334a731073af292330547d63b/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L260)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> we are *highly suggest*

Would change to "we recommend".
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ahie** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 260 at r4](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WRDsN-eR8fiS0cp_H:-L1WS9ud6mXgTL4wfp9a:b-14augc) ([raw file](https://github.com/vaadin/flow/blob/2d38ad3575525ce334a731073af292330547d63b/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L260)):*
<details><summary><i>Previously, ZheSun88 (Sun Zhe) wrote…</i></summary><blockquote>

done
</blockquote></details>

And perhaps leave out the emphasis :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ZheSun88** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[flow-documentation/application-structure/tutorial-bootstrap.asciidoc, line 260 at r4](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WRDsN-eR8fiS0cp_H:-L1WUCWbExxmO6udrl7H:b-okqj9m) ([raw file](https://github.com/vaadin/flow/blob/2d38ad3575525ce334a731073af292330547d63b/flow-documentation/application-structure/tutorial-bootstrap.asciidoc#L260)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

And perhaps leave out the emphasis :)
</blockquote></details>

umm removed.. but doesn't it matter?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ahie** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[flow-documentation/src/main/java/com/vaadin/flow/tutorial/BootstrapPage.java, line 141 at r5](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WUZ_DhF5O0vUX15jd:-L1WUZ_DhF5O0vUX15je:bqy4ycv) ([raw file](https://github.com/vaadin/flow/blob/7df7827a567442db19a2ae5afaa2459824fa4f2b/flow-documentation/src/main/java/com/vaadin/flow/tutorial/BootstrapPage.java#L141)):*
> favicon.html

Sorry, didn't notice this. build will fail because this will need to be updated to `your-content.html` as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ZheSun88** [comment]
Review status: 1 of 2 files reviewed at latest revision, 4 unresolved discussions.

---

*[flow-documentation/src/main/java/com/vaadin/flow/tutorial/BootstrapPage.java, line 141 at r5](https://reviewable.io:443/reviews/vaadin/flow/3207#-L1WUZ_DhF5O0vUX15jd:-L1WZ7XN6lZ3-M6wBx9t:b-aakle2) ([raw file](https://github.com/vaadin/flow/blob/7df7827a567442db19a2ae5afaa2459824fa4f2b/flow-documentation/src/main/java/com/vaadin/flow/tutorial/BootstrapPage.java#L141)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> favicon.html

Sorry, didn't notice this. build will fail because this will need to be updated to `your-content.html` as well.
</blockquote></details>

oh, sorry, my fault.. i have totally forgotten that 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*

**Comment by ahie** [comment]
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3207)*
